app/modules/ads/ml/tb-viewability.js

```diff
@@ -0,0 +1,73 @@
+let config = {};
+
+export const tbViewability = {
+  configureTbViewability(billTheLizardConfig) {
+    const {
+      context,
+      ViewabilityCounter,
+    } = window.Wikia.adEngine;
+    const { billTheLizard } = window.Wikia.adServices;
+    const viewabilityCounter = ViewabilityCounter.make();
+    config = billTheLizardConfig;
+
+    if (!this.hasAvailableModels(config, 'tb_viewability')) {
+      return;
+    }
+
+    billTheLizard.projectsHandler.enable('tb_viewability');
+
+    // viewabilityCounter already sets default to 0.5 and is between 0-1
+    context.set('services.billTheLizard.parameters', {
+      tb_viewability: {
+        session_viewability_all: viewabilityCounter.getViewability(),
+        session_viewability_icb: viewabilityCounter.getViewability('incontent_boxad'),
+        session_viewability_tb: viewabilityCounter.getViewability('top_boxad'),
+        ref: context.get('targeting.ref') || null,
+        scroll_y: this.calculateScrollY(),
+        session_scroll_speed: this.calculateSessionScrollSpeed() || 0,
+        s0v: context.get('targeting.s0v') || null,
+        s2: context.get('targeting.s2') || null,
+      },
+    });
+
+    // consulted 'top_page' parameter with Martyna, if we sent top_boxad cheshire is going mad
+    billTheLizard.call(['tb_viewability'], 'top_page');
+  },
+
+  calculateScrollY() {
+    const scrollY = (window.scrollY || window.pageYOffset || 0) / 4500;
+
+    return this.resultsProcessor(scrollY);
+  },
+
+  calculateSessionScrollSpeed() {
+    const { ScrollSpeedCalculator } = window.Wikia.adServices;
+    const scrollSpeedCalculator = ScrollSpeedCalculator.make();
+    const scrollSpeed = (scrollSpeedCalculator.getAverageSessionScrollSpeed()) / 1000;
+
+    return this.resultsProcessor(scrollSpeed);
+  },
+
+  hasAvailableModels(btlConfig, projectName) {
```

Comment:
This function is the same in `tb-viewability.js` and `cheshire-cat.js`. Can we move it to a shared part?

Reply:
Moved it to utils.js file. For now there's only one function but maybe with more models there will be more